beta/src/components/MDX/Sandpack/Preview.tsx

```diff
@@ -136,9 +126,14 @@ export function Preview({
       }}>
       <div
         className={cn(
-          'p-0 sm:p-2 md:p-4 lg:p-8 bg-card dark:bg-wash-dark h-full relative rounded-b-lg lg:rounded-b-none'
-        )}
-        style={{overflow}}>
+          'p-0 sm:p-2 md:p-4 lg:p-8 bg-card dark:bg-wash-dark h-full relative rounded-b-lg lg:rounded-b-none',
+          // Allow content to be scrolled if it's too high to fit.
+          // Note we don't want this in the expanded state
+          // because it breaks position: sticky (and isn't needed anyway).
+          // We also don't want this for errors because they expand
+          // parent and making them scrollable is confusing.
+          !isExpanded && !error && isReady ? 'overflow-auto' : null
```

Comment:
Please review changes in this file very carefully and make sure you don’t break any styles.